vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/ReasonerImpl.java

```diff
@@ -169,21 +170,21 @@ private String edbPredicatesConfigToString() {
 	}
 
 	private void loadInMemoryFacts() throws EDBConfigurationException {
-		final Map<String, List<List<Term>>> factsMap = new HashMap<>();
+		final Map<Predicate, List<List<Term>>> factsMap = new HashMap<>();
 		for (final Atom fact : this.facts) {
 			factsMap.putIfAbsent(fact.getPredicate(), new ArrayList<>());
 			factsMap.get(fact.getPredicate()).add(fact.getTerms());
 		}
-		for (final String predName : factsMap.keySet()) {
-			final List<List<Term>> predNameArgs = factsMap.get(predName);
-			final int arity = predNameArgs.get(0).size();
-			final String[][] tuplesMatrix = new String[predNameArgs.size()][arity];
-			for (int i = 0; i < predNameArgs.size(); i++) {
+		for (final Predicate pred : factsMap.keySet()) {
```

Comment:
style: call variables like their class, i.e., "predicate" instead of "pred"

Reply:
I will fix this in the "query" branch